### DISCUSSION TITLE
IFrame 组件 嵌套网页--加一个参数用于设置首选语言（达到UI与嵌套的网页语言一致）

### DISCUSSION THREAD

**Discussion by ethanchendevs** [dotnetcore/BootstrapBlazor#3667] [Q&A] [answered]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Is your feature request related to a problem? Please describe the problem.

使用场景说明
服务端已部署有网站---url+?culture=zh-cn 切换语言--直接用浏览器切换可以生效（允许刷新），在组件中无法生效
网站范例地址：[http://localhost:5210/Demo/function1?culture=zh-cn](https://gitee.com/link?target=http%3A%2F%2Flocalhost%3A5210%2FDemo%2Ffunction1%3Fculture%3Dzh-cn)
在Blazor Page页面 添加IFrame 组件给SRC赋值网址+?culture=zh-cn 切换语言--UI画面切换语言OK，但IFrame组件中的页面有观察到语言切换为目标语言后又很快自动刷新为非目标语言(后面看了大神的视频才了解到还有浏览器上设定的语言在起作用)

### Describe the solution you'd like

IFrame 组件 嵌套网页--加一个参数用于设置首选语言（达到UI与嵌套的网页语言同步切换且不刷新页面不受影响）
或是有其它方法可以在Blazor 中嵌第三方网页可以同步切换语言
感谢大佬！

### Additional context

直接用浏览器访问切换地址-中文
![image](https://github.com/dotnetcore/BootstrapBlazor/assets/142855022/365fa49e-025c-4272-b28a-0087bc892cb2)
直接用浏览器访问切换地址-英文
![image](https://github.com/dotnetcore/BootstrapBlazor/assets/142855022/6d765696-700d-4962-9ea7-d00883786d94)
浏览器首选语言
![image](https://github.com/dotnetcore/BootstrapBlazor/assets/142855022/dacfc6d8-b378-4550-8a4a-e522f847d11e)

异常图示1
![image](https://github.com/dotnetcore/BootstrapBlazor/assets/142855022/8612bc0b-4b4a-49f1-b8d0-f8d540427c76)
异常图示2
![image](https://github.com/dotnetcore/BootstrapBlazor/assets/142855022/a56e0de8-3db7-4f5c-b5f0-6713fbb9fc38)
前台代码
![image](https://github.com/dotnetcore/BootstrapBlazor/assets/142855022/d5a49bc7-7fa6-4804-87ad-8233711311bf)

C#代码
![image](https://github.com/dotnetcore/BootstrapBlazor/assets/142855022/ea0ad041-1d30-49e6-89dd-881976758c34)

**Comment by bb-auto** [comment]
@aegitdev2023 Thank you for reporting. We will give triage later.

**Comment by densen2014** [comment]
直接把官网文档站点套进去一样吗

**Comment by ethanchendevs** [comment]
> 直接把官网文档站点套进去一样吗
感谢大佬指点:
刚试了用官网地址：https://www.blazor.zone/components 测试无法访问看不到页面--提示拒绝连接（应该是站点有屏蔽嵌入式访问页面）
BootstrapBlazor官网地址:
![image](https://github.com/dotnetcore/BootstrapBlazor/assets/142855022/f837f0d2-1116-42e2-93c3-8eae04ab64bf)
微软官网--测试：
![image](https://github.com/dotnetcore/BootstrapBlazor/assets/142855022/a13250ed-ffcb-4c91-8f2e-5dce92c2c8bb)

**Reply by ArgoZhang** [reply]
@aegitdev2023 感谢提交 Issue,能否帮忙右上角点一下 star

如果 `IFrame` 内容文化信息需要与主站一致,最合适的方法使 `IFrame` 组件的 `Src` 参数的 Url 增加当前期望文化信息

`<IFrame Src="www.domain.com/index?culture=zh-CN" />`

类似这种

**Reply by ethanchendevs** [reply]
您好：
主站在切换语言时有同步传入切换组件中URL+?culture=zh-CN的操作：目前验证出可以让组件中的嵌套页面切换语言是通过更改浏览器的首选语言才会随着变化;
![image](https://github.com/dotnetcore/BootstrapBlazor/assets/142855022/1496aec0-de50-48c3-a7b9-b602f1c40e1f)

验证视频

https://github.com/dotnetcore/BootstrapBlazor/assets/142855022/89f87139-c86f-4e00-aadc-919f5b7a3a0f

Demo项目已经上传如下地址--有兴趣可以研究看看，感谢
https://github.com/aegitdev2023/Public/tree/main/BootstrapBlazor.IFrameDemo

**Reply by densen2014** [reply]
复现步骤是？望提供一个直接能复现的demo

视频好像打不开

![image](https://github.com/dotnetcore/BootstrapBlazor/assets/8428709/b45cebcd-eae4-42a1-b967-1655b7a99d66)

**Reply by ethanchendevs** [reply, marked as answer]
微软官网了解到这个参数（ApplyCurrentCultureToResponseHeaders）设为true后嵌入组件中的页面可以跟随主站的语言自动切换
![image](https://github.com/dotnetcore/BootstrapBlazor/assets/142855022/a5406422-1787-4423-9334-1a2a272d703d)
Program.cs 添加属性后发布
![image](https://github.com/dotnetcore/BootstrapBlazor/assets/142855022/1fe27fc7-ffb6-4639-98ff-b07d2ad16db4)
将网站挂在IIS上模拟测试
测试结果图示 繁体
![image](https://github.com/dotnetcore/BootstrapBlazor/assets/142855022/0017efb2-324f-4b6b-b052-95a2df8af91b)
测试结果图示 简体
![image](https://github.com/dotnetcore/BootstrapBlazor/assets/142855022/b07c9cd2-affb-4c75-b5ec-3ef565ca7d67)
测试结果图示 英文
![image](https://github.com/dotnetcore/BootstrapBlazor/assets/142855022/0997134e-b60c-492b-88e6-252dc5d6f6ff)
此问题已解决，感谢大佬的提示！
